lib/apiservers/service/restapi/handlers/vch_log_get.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/go-openapi/runtime/middleware"
+
+	"github.com/vmware/vic/lib/apiservers/service/restapi/handlers/util"
+	"github.com/vmware/vic/lib/apiservers/service/restapi/operations"
+	"github.com/vmware/vic/lib/install/data"
+	"github.com/vmware/vic/lib/install/management"
+	"github.com/vmware/vic/lib/install/validate"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/vsphere/datastore"
+)
+
+const (
+	logFilePrefix = "vic-machine" // logFilePrefix is the prefix for file names of all vic-machine log files
+	logFileSuffix = ".log"        // logFileSuffix is the suffix for file names of all vic-machine log files
+)
+
+// VCHLogGet is the handler for getting the log messages for a VCH
+type VCHLogGet struct {
+}
+
+// VCHDatacenterLogGet is the handler for getting the log messages for a VCH within a Datacenter
+type VCHDatacenterLogGet struct {
+}
+
+func (h *VCHLogGet) Handle(params operations.GetTargetTargetVchVchIDLogParams, principal interface{}) middleware.Responder {
+	op := trace.NewOperation(params.HTTPRequest.Context(), "VCHLogGet: %s", params.VchID)
+
+	b := buildDataParams{
+		target:     params.Target,
+		thumbprint: params.Thumbprint,
+	}
+
+	d, err := buildData(op, b, principal)
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	d.ID = params.VchID
+
+	helper, err := getDatastoreHelper(op, d)
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	logFilePaths, err := getAllLogFilePaths(op, helper)
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	output, err := getContentFromLogFiles(op, helper, logFilePaths)
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	return operations.NewGetTargetTargetVchVchIDLogOK().WithPayload(output)
```

Comment:
I take it this is NOT expected to stream the log? Is it expected to block or truncate at whatever is the end-of-file when we're reading it?

Reply:
Correct. The proposal is to eventually use a more appropriate protocol (WebSockets?) to allow for streaming of logs. (See #6702 for more information.)